src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,125 @@
+        # TODO: if more generic approach will be required we'll need to switch to
+#         super(MSPileup, self).__init__(msConfig, **kwargs)
+
+        # for authz we should provide relevant section in MS configuration
+        # if no role/group is provided then authz_match used in HTTP APIs
+        # will always return valid state. Therefore, for test purposes we can
+        # use empty role and group, while for production setup we should set thenm up
+        authzDefaults = msConfig.get('authz_defaults', {})
+        self.role = authzDefaults.get('role', [])
+        self.group = authzDefaults.get('group', [])
+        self.mgr = MSPileupData(msConfig)
+
+    def status(self):
+        """
+        Provide MSPileup status API. We should extend it to check DB connection, etc.
+
+        :return: status dictionary
+        """
+        summary = dict(PILEUP_REPORT)
+        summary.update({'thread_id': current_thread().name})
+        return summary
+
+    def get(self, **kwargs):
+        """
+        Provide MSPileup get API
+
+        :param **kwargs: provide key=value (or spec) input
+        :return: results of MSPileup data layer (list of dicts)
+        """
+        spec = {}  # (get all full docs) retrieve a list with the full documentation of all the pileups
+        if 'spec' in kwargs:
+            # use specific spec (JSON query)
+            spec = kwargs['spec']
+        elif 'pileupName' in kwargs:
+            # get docs for given pileupName
+            spec = {'pileupName': kwargs['pileupName']}
+        elif 'campaign' in kwargs:
+            # get docs for given campaign
+            spec = {'campaign': kwargs['campaign']}
+        else:
+            for key, val in kwargs.items():
+                spec[key] = val
+
+        # check if filters are present and use it as projection fields
+        projection = {}
+        for key in kwargs.get('filters', []):
+            projection[key] = 1
+        results = self.mgr.getPileup(spec, projection)
+
+        return results
+
+    def query(self, spec, projection=None):
+        """
+        Query POST HTTP request API
+
+        :param spec: provide query JSON spec to MSPileup data layer
+        :return: results of MSPileup data layer (list of dicts)
+        """
+        return self.mgr.getPileup(spec, projection)
+
+    def create(self, pdict):
+        """
+        Create POST HTTP request API
+
+        :param pdict: input MSPilup data dictionary
+        :return: results of MSPileup data layer (list of dicts)
+        """
+        authz_match(self.role, self.group)
+        return self.mgr.createPileup(pdict)
+
+    def update(self, pdict):
+        """
+        Update PUT HTTP request API
+
+        :param pdict: input MSPilup data dictionary
+        :return: results of MSPileup data layer (list of dicts)
+        """
+        authz_match(self.role, self.group)
+        return self.mgr.updatePileup(pdict)
+
+    def delete(self, spec):
```

Comment:
Lastly, I'd rename it to `deletePileup`. Same comment for the docstring.

Reply:
done